app/models/coin.rb

```diff
@@ -184,6 +184,21 @@ def market_info(market_data = nil)
     data
   end
 
+  def token_metrics
```

Comment:
Hm, from my understanding, this method will make 10 GET requests on a cold cache, will that be a performance problem?

Reply:
Hasn't seemed to be an issue. We should definitely look to improving how we fetch from cache as the amount of data we get comes from cache. But, compared to the initial 10 remote calls, I think it's been pretty trivial.

We can try leveraging read_multi, but we can't just replace it directly since we'll have to also check/fetch any missing keys afterwards.